src/DomainNetMap.cpp

```diff
@@ -0,0 +1,344 @@
+/*
+ * Copyright (c) 2015 - 2023, Intel Corporation
+ * SPDX-License-Identifier: BSD-3-Clause
+ */
+
+#include "DomainNetMapImp.hpp"
+
+#include <cmath>
+#include <fstream>
+
+#include "geopm/PlatformIO.hpp"
+#include "geopm/Exception.hpp"
+#include "geopm/Helper.hpp"
+#include "NNFactoryImp.hpp"
+
+namespace geopm
+{
+    const std::vector<std::string> DomainNetMapImp::M_EXPECTED_KEYS = {
+            "layers",
+            "signal_inputs",
+            "delta_inputs",
+            "trace_outputs",
+            "description"};
+
+    std::unique_ptr<DomainNetMap> DomainNetMap::make_unique(const std::string &nn_path,
+                                                            geopm_domain_e domain_type,
+                                                            int domain_index)
+    {
+        return geopm::make_unique<DomainNetMapImp>(nn_path, domain_type, domain_index);
+    }
+
+    std::shared_ptr<DomainNetMap> DomainNetMap::make_shared(const std::string &nn_path,
+                                                            geopm_domain_e domain_type,
+                                                            int domain_index)
+    {
+        return std::make_shared<DomainNetMapImp>(nn_path, domain_type, domain_index);
+    }
+
+    DomainNetMapImp::DomainNetMapImp(
+            const std::string &nn_path,
+            geopm_domain_e domain_type,
+            int domain_index)
+        : DomainNetMapImp(nn_path, domain_type, domain_index, platform_io(),
+                          NNFactory::make_shared()) {
+    }
```

Comment:
This is the style we use for classes with passed in paramaters and initialization lists, and empty constructor bodies:
```suggestion
    DomainNetMapImp::DomainNetMapImp(const std::string &nn_path,
                                     geopm_domain_e domain_type,
                                     int domain_index)
        : DomainNetMapImp(nn_path, domain_type, domain_index, platform_io(),
                          NNFactory::make_shared())
    {
    
    }
```